src/NServiceBus.Transport.SqlServer/Receiving/ProcessStrategy.cs

```diff
@@ -99,7 +99,13 @@ protected async Task<bool> TryHandleDelayedMessage(Message message, SqlConnectio
             }
             catch (QueueNotFoundException e)
             {
-                log.Error($"Message {message.TransportId} cannot be forwarded to its destination queue {e.Queue} because it does not exist.");
+                var hasEnclosedMessageTypeHeader = message.Headers.TryGetValue(Headers.EnclosedMessageTypes,
+                    out var enclosedMessageTypeHeader);
+
+                log.Error(
+                    hasEnclosedMessageTypeHeader
+                        ? $"Message {message.TransportId} of type {enclosedMessageTypeHeader} cannot be forwarded to its destination queue {e.Queue} because it does not exist."
+                        : $"Message {message.TransportId} cannot be forwarded to its destination queue {e.Queue} because it does not exist.");
```

Comment:
```suggestion
                log.Error(
                    hasEnclosedMessageTypeHeader
                        ? $"Message with ID '{message.TransportId}' of type '{enclosedMessageTypeHeader}' cannot be forwarded to its destination queue '{e.Queue}' because it does not exist."
                        : $"Message with ID '{message.TransportId}' cannot be forwarded to its destination queue '{e.Queue}' because it does not exist.");
```

but given that we are touching this log statement I wonder if we should tweak it to be compatible with structured logging by switching to the overload that has formatter and args?